drivers/i2c/i2c_shell.c

```diff
@@ -87,6 +91,130 @@ static int cmd_i2c_recover(const struct shell *shell,
 	return 0;
 }
 
+/* i2c write <device> <dev_addr> [<byte1>, ...] */
+static int cmd_i2c_write(const struct shell *shell, size_t argc, char **argv)
+{
+	u8_t buf[MAX_I2C_BYTES];
+	struct device *dev;
+	int num_bytes;
+	int reg_addr;
+	int dev_addr;
+	int i;
+
+	dev = device_get_binding(argv[1]);
+	if (!dev) {
+		shell_error(shell, "I2C: Device driver %s not found.", argv[1]);
+		return -ENODEV;
+	}
+
+	dev_addr = strtol(argv[2], NULL, 16);
+	reg_addr = strtol(argv[3], NULL, 16);
+	num_bytes = argc - 4;
+	if (num_bytes < 0)
+		return 0;
+	if (num_bytes > MAX_I2C_BYTES)
```

Comment:
need curly braces around if statement

Reply:
Is there a way to check for this? Checkpatch doesn't warn me and I'm not used to it as U-Boot, kernel, etc. are different.

Reply:
checkpatch should definitely warn you about this. Maybe there's a bug in it.